### ISSUE TITLE
Automated workflow to add Quenya, Telerin (etc.) cognates to existing entries?

### ISSUE THREAD

**Issue by L-ND-L** [Omikhleia/sindict#51]
Here in English, as it's a more generic issue:

Wouldn't it be possible to develop a workflow allowing easy, structured referencing of cognate words in Quenya and Telerin (and maybe a few more languages: Nandorin, Danian, Ilkorin...)?

This would then allow for simple acceptance / rejection (maybe with comment) from the project administrator(s).

Once developed (if possible / realistic / useful), this would be easier and faster to manage the addition of cognates than the generic issue reporting, which requires further manual processing to add the cognates to the database.

**Comment by Omikhleia** [comment]
On one hand, adding cognates for all languages would be a _monumental_ task - of a kind this project never touched since its inception more than 20 years ago... So a first question could be: is it worth involving time in this direction _now_? If we start with cognates, we'll even want roots, which changed over Tolkien's lifetime - that's a lot of things to properly encode...

I only know _one_ project that sort of achieved it: [Eldamo](https://eldamo.org/)... So a second question could be: what would we solve here differently, for the effort to be worth of it?

As for developing things, let's face it, there was always a lack of long-term contributors, not to say developers.

Regarding developers, 
- I've already walked on that path in 2011-2012 with a tool that was never used... Remember the [Ladon experiment](https://forum.tolkiendil.com/thread-2851-post-112671.html#pid111593)? I gave up when Firefox changed its add-ons structure, but anyway, it had the taste of a failure (1).
- Other that that, we never saw anything similar to Dragon Flame or Hesperides in more than 10+ years...
- As of "online" user-friendly collaborative tools, there's of course [Parf Edhellen](https://www.elfdict.com/). I let you make your opinion, it's of a too different nature compared to the things we did here.

Regarding contributors... I am afraid I don't think it's realistic, unless you have a lot (and a whole lot) of free time. Those who could contribute, in a way, preferred doing their own things and even to sell them (I'm sure you know [this one](https://www.amazon.fr/Dictionnaire-Elfique-Sindarin-Fran%C3%A7ais-Fran%C3%A7ais-Sindarin-Pocket/dp/1508774773) and [this other one](https://www.amazon.fr/Sindarin-English-English-Sindarin-Dictionary-J-M-Carpenter/dp/1291332162) for an example. All derivatives, in a way or another, of this very project. All failing short, if I may say...)

This dictionary was once one of the best of its kind - if not the best -, I don't think saying that is so pretentious. Nowadays, though, one cannot do as if Eldamo didn't existed ;-) Yet, this dictionary can _still_ count, despite its age. But asking for more than that... I don't know.

I'm sorry to sound quite negative here. Find developers (and for such a workflow, you mean coders(2), right?), and I'll gladly reconsider my position.

(1) It's partly lost now. Bits were recovered, though I have no idea if it could be revived without lots of tweaks. But the video link above was cool...

(2) Just in case I am not clear here: I don't consider the lexicon to be "code". It's just tagged information (ok, in XML, but heh). A big text file with mark-up. The code, for me, is what transforms it to HTML, PDF, or whatever.

**Comment by Omikhleia** [comment]
I had even forgotten the other [Ladon experiment](https://www.youtube.com/watch?v=O2IEp7akYoQ) video. Showing the main dictionary tool (not the editor). Makes me feel even older :stuck_out_tongue:

**Comment by L-ND-L** [comment]
> On one hand, adding cognates for all languages would be a _monumental_ task - of a kind this project never touched since its inception more than 20 years ago... So a first question could be: is it worth involving time in this direction _now_? If we start with cognates, we'll even want roots, which changed over Tolkien's lifetime - that's a lot of things to properly encode...

You're right. On the other hand, there are already some entries with Q(u)enya and Telerin cognates, and it would be sad to remove them. So for the sake of consistency, one may want to continue adding cognates to existing entries that don't have them yet. Cognates for other languages would definitely be an additional task, albeit a limited one (basically covering only _The Etymologies_).

As for P.Q. / C.E. words, and roots, that's even a further step. It would definitely be nice to be able to encode them properly (even if we don't start adding them), but I agree this goes well beyond the initial scope of the Dictionary, and would probably overlap more and more with Eldamo.

> I only know _one_ project that sort of achieved it: [Eldamo](https://eldamo.org/)... So a second question could be: what would we solve here differently, for the effort to be worth of it?

This brings us to your second question, which is a very good one, obviously. I do have three arguments in favor of doing things differently, and one against. Pros: (1) there are a fair number of cases where I have good reasons to think that Paul got some interpretations wrong, and it takes way too much efforts to convince him, even with a lot of evidence; (2) there is no French version of Eldamo; (3) developing one would soon create additional issues, as Paul doesn't provide a way for others to track his ongoing changes. Cons: I don't have the time to do it now, and won't for the short to medium terms.

> Regarding contributors... I am afraid I don't think it's realistic, unless you have a lot (and a whole lot) of free time. Those who could contribute, in a way, preferred doing their own things and even to sell them (I'm sure you know [this one](https://www.amazon.fr/Dictionnaire-Elfique-Sindarin-Fran%C3%A7ais-Fran%C3%A7ais-Sindarin-Pocket/dp/1508774773) and [this other one](https://www.amazon.fr/Sindarin-English-English-Sindarin-Dictionary-J-M-Carpenter/dp/1291332162) for an example. All derivatives, in a way or another, of this very project. All failing short, if I may say...)

You may indeed. :)

> I'm sorry to sound quite negative here. Find developers (and for such a workflow, you mean coders(2), right?), and I'll gladly reconsider my position.

I do mean developers, and of course I don't have names right now. If I did, there is a large Tolkienian French website whose overhaul is now long overdue, and I would direct them there as a priority. ;)
 
> (2) Just in case I am not clear here: I don't consider the lexicon to be "code". It's just tagged information (ok, in XML, but heh). A big text file with mark-up. The code, for me, is what transforms it to HTML, PDF, or whatever.

I actually tried to open the xml file last week, but my computer refused to comply... I'll have to check why when I have time to do so.

**Comment by Omikhleia** [comment]
Greetings.
This issue has been inactive for several years, without any proposed resolution.

> one may want to continue adding cognates to existing entries that don't have them yet. 

This is a contributive project on GitHub, by definition contributions are welcome.
I made it clear in 2021 that I wouldn't do any effort on my side, but by definition "pull requests" are welcome, if carried with a bit of methodology and appropriate development regarding the tooling (for PDF or online web version generation)

But it also does seem clear that this dictionary is focused on Sindarin, without any contributor to extend it beyond that scope for years.

So I am setting the "pending closure" label on this issue, as part of a backlog cleaning pass.

Please feel free to comment, you or other contributors, as a last change to eventually say their word here. Otherwise, after a short period of time, this issue will be closed without further notice.
